Gruppe.A1.End-to-End_Encryption_mit_IPv6_und_integriertem_IPSec.ThushjandanPonnudurai.PascalBaumann.Termpaper.tex

```diff
@@ -62,12 +73,103 @@ \chapter{Theoretical part}
+A datagram is composed of an IP header and a payload. Source and destination IP addresses and other meta data are parts of the IP header and are required to deliver a datagram. The payload includes data that is transported.
+
+IPv4 is the first major version and is at the moment the dominant protocol version of the Internet. Due to a lack of IPv4 addresses the successor IPv6 was born. \parencite{NadeemUnuth2016}
 
 \subsection{IPv4}
 \label{ssec:IPv4}
+Internet Protocol version 4 (IPv4) is specified in the IETF RFC 791 document and is existing since 1980. To route a packet across the networks, every host in the network has a logical address, which is the IPv4 address. This IPv4 addressing system is based on a 32-bit logical address, which amounts to 4'294'967'296 unique addresses.
+
+An example of an IPv4 address is "148.21.45.110". The address is written in dot decimal notation and it has four octets of 8-bits. The binary form of this address will be 10010100.00010101.00101101.01101110. \parencite{Babatunde2014}
 
+At the moment there is an IPv4 address exhaustion problem. Since the 3th of February 2011 "Internet Corporation for Assigned Names and Numbers" (ICANN) doesn't have any free blocks of IPv4 addresses. Due this problem IPv6 was developed and everything will be migrated in IPv6 over the long-term. 
 \subsection{IPv6}
 \label{ssec:IPv6}
+Internet Protocol version 6 (IPv6) is the successor of IPv4. It was specified in 1998. The growth of the Internet led to a need for new alternatives. The problem is that IPv4 cannot provide the needed number of logical addresses. For that reason IPv6 protocol was developed.\parencite[11]{Babatunde2014}
+
+Several important areas were enhanced in the IPv6 protocol:
+\begin{itemize}
+	\item Expanded addressing
+	\item Simplified header format
+	\item Improved extension and option support
+\end{itemize}
+With the expanded address space the issue with the lack of addresses will be eliminated. Also the addressing architecture and the simplified header format are improving the routing efficiency. Fewer packets needs to be processed thanks to the simplified header format. 
+Every IPv6 packet has a fixed header length of 40 octets. But any options can be appended in extension headers after the IPv6 main header. \parencite[106-107,123-124]{Loshin2004}
+
+\subsubsection{IPv6 Addressing}
+\label{sssec:ipv6:addressing}
+There are three types of IPv6 addresses:
+\begin{itemize}
+	\item Unicast. A packet sent to an unicast address is delivered to the interface identified by that address.
+	\item Multicast. A packet sent to a multicast address is delivered to all interfaces identified by that address. This address could be existing on different hosts.
+	\item Anycast. A packet sent to a anycast address is delivered to one of the interfaces identified by that address. This address could be existing on different hosts.
+\end{itemize}
+Broadcast is not existing anymore in IPv6. \parencite[142-143]{Loshin2004}
```

Comment:
The traditional IPv6 broadcast does not exist anymore, instead the functionality is provided by specific multicast addresses. To find a router for example, one can ping FF02::2, as any router interface will subscribe to this multicast address.